In [4]:
# Exploring a dataset of gun deaths in the USA
import csv
f = open("guns.csv", 'r')
dataset = csv.reader(f)
data = list(dataset)
print(data[0:5])

[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education'], ['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]


In [5]:
# Splitting the header from the rest of the data
headers = data[0]
data = data[1:]
print(headers)
print(data[0:5])

['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education']
[['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', '2']]


In [6]:
# Make a dictionary for every year in the dataset, and count all the deaths per year
years = [col[1] for col in data]
year_counts = {}
for row in years:
    if row not in year_counts:
        year_counts[row] = 1
    else:
        year_counts[row] += 1
print(year_counts)

{'2013': 33636, '2014': 33599, '2012': 33563}


In [7]:
# Transforming the data about the month into a usable format to then explore further
import datetime
dates = [datetime.datetime(year=int(row[1]), month=int(row[2]), day=1) for row in data]
print(dates[:5])

[datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 1, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0), datetime.datetime(2012, 2, 1, 0, 0)]


In [16]:
# Display the number of deaths for each month from 2012 to 2014
date_counts = {}
for row in dates:
    if row not in date_counts:
        date_counts[row] = 1
    date_counts[row] += 1
print(date_counts)

{datetime.datetime(2012, 7, 1, 0, 0): 3027, datetime.datetime(2013, 7, 1, 0, 0): 3080, datetime.datetime(2014, 12, 1, 0, 0): 2858, datetime.datetime(2013, 12, 1, 0, 0): 2766, datetime.datetime(2013, 3, 1, 0, 0): 2863, datetime.datetime(2014, 9, 1, 0, 0): 2915, datetime.datetime(2012, 8, 1, 0, 0): 2955, datetime.datetime(2013, 2, 1, 0, 0): 2376, datetime.datetime(2012, 1, 1, 0, 0): 2759, datetime.datetime(2013, 9, 1, 0, 0): 2743, datetime.datetime(2013, 1, 1, 0, 0): 2865, datetime.datetime(2014, 6, 1, 0, 0): 2932, datetime.datetime(2013, 6, 1, 0, 0): 2921, datetime.datetime(2014, 4, 1, 0, 0): 2863, datetime.datetime(2012, 12, 1, 0, 0): 2792, datetime.datetime(2014, 5, 1, 0, 0): 2865, datetime.datetime(2012, 3, 1, 0, 0): 2744, datetime.datetime(2014, 11, 1, 0, 0): 2757, datetime.datetime(2012, 11, 1, 0, 0): 2730, datetime.datetime(2014, 3, 1, 0, 0): 2685, datetime.datetime(2013, 5, 1, 0, 0): 2807, datetime.datetime(2013, 10, 1, 0, 0): 2809, datetime.datetime(2012, 9, 1, 0, 0): 2853, date

In [8]:
# Create dictionary for gun deaths by gender
sexes = [row[5] for row in data]
sex_counts = {}
for row in sexes:
    if row not in sex_counts:
        sex_counts[row] = 1
    sex_counts[row] += 1
print(sex_counts)

{'F': 14450, 'M': 86350}


In [9]:
# Create dictionary for gun deaths by race
races = [row[7] for row in data]
race_counts = {}
for row in races:
    if row not in race_counts:
        race_counts[row] = 1
    race_counts[row] += 1
print(race_counts)

{'White': 66238, 'Hispanic': 9023, 'Native American/Native Alaskan': 918, 'Asian/Pacific Islander': 1327, 'Black': 23297}


It is clear that most gun deaths are happening to Males, and in terms of race, most are White, followed by Blacks, and Hispanics.

In [10]:
# Importing some census data in order to further our analysis
import csv
f = open("census.csv", 'r')
census = list(csv.reader(f))
print(census)

[['Id', 'Year', 'Id', 'Sex', 'Id', 'Hispanic Origin', 'Id', 'Id2', 'Geography', 'Total', 'Race Alone - White', 'Race Alone - Hispanic', 'Race Alone - Black or African American', 'Race Alone - American Indian and Alaska Native', 'Race Alone - Asian', 'Race Alone - Native Hawaiian and Other Pacific Islander', 'Two or More Races'], ['cen42010', 'April 1, 2010 Census', 'totsex', 'Both Sexes', 'tothisp', 'Total', '0100000US', '', 'United States', '308745538', '197318956', '44618105', '40250635', '3739506', '15159516', '674625', '6984195']]


In [11]:
# Creating a dictionary mapping the population for each race category
mapping = {
    "Asian/Pacific Islander": 15159516 + 674625,
    "Native American/Native Alaskan": 3739506,
    "Black": 40250635,
    "Hispanic": 44618105,
    "White": 197318956
}

# Calculating the number of deaths per 100,000 people for each race
race_per_hundredk = {}
for k,v in race_counts.items():
    race_per_hundredk[k] = (v / mapping[k]) * 100000

race_per_hundredk

{'Asian/Pacific Islander': 8.380625131480135,
 'Black': 57.87983220637389,
 'Hispanic': 20.222732453563413,
 'Native American/Native Alaskan': 24.548697073891578,
 'White': 33.568999827872595}

In [12]:
# Create a dictionary of the number of homicides per race category
intents = [row[3] for row in data]
races = [row[7] for row in data]
homicide_race_counts = {}
for i,race in enumerate(races):
    if race not in homicide_race_counts:
        homicide_race_counts[race] = 0
    if intents[i] == "Homicide":
        homicide_race_counts[race] += 1
print(homicide_race_counts)   
        

{'White': 9147, 'Hispanic': 5634, 'Native American/Native Alaskan': 326, 'Asian/Pacific Islander': 559, 'Black': 19510}


In [13]:
# Calculating the homicide rate per 100,000 for each race
race_per_hundredk = {}
for k,v in homicide_race_counts.items():
    race_per_hundredk[k] = (v / mapping[k]) * 100000

race_per_hundredk

{'Asian/Pacific Islander': 3.530346230970155,
 'Black': 48.471284987180944,
 'Hispanic': 12.627161104219914,
 'Native American/Native Alaskan': 8.717729026240365,
 'White': 4.6356417981453335}

Looking at the homicide rate per 100,000 people, it is clear that Blacks have a disproportionate homicide rate, as compared to other races.  It is 4 times higher than the second highest rate, which is for Hispanics.

In [23]:
# Exploring the number of gun deaths by education level

education = [row[10] for row in data]
education_counts = {}
for row in education:
    if row not in education_counts:
        education_counts[row] = 1
    education_counts[row] += 1
print(education_counts)
low_education = education_counts["1"] + education_counts["2"] + education_counts["3"]  
print(low_education)
high_education = education_counts["4"] + education_counts["5"] 
print(high_education)

{'3': 21681, '5': 1370, '4': 12947, '1': 21824, 'NA': 54, '2': 42928}
86433
14317


There are almost 6 times more gun deaths of low educated people than there are from high educated people.

In [27]:
# Exploring the number of gun deaths by location

location = [row[9] for row in data]
location_counts = {}
for row in location:
    if row not in location_counts:
        location_counts[row] = 1
    else:
        location_counts[row] += 1
print(location_counts)
home_deaths_ratio = location_counts["Home"] / len(data)
print(home_deaths_ratio)

{'NA': 1384, 'Home': 60486, 'Industrial/construction': 248, 'Other specified': 13751, 'Other unspecified': 8867, 'Street': 11151, 'Farm': 470, 'Trade/service area': 3439, 'Sports': 128, 'Residential institution': 203, 'School/instiution': 671}
0.6000714299886902


Most gun deaths happen at home, according to this data, 60% to be exact.

In [29]:
# An easy way to calculate the number of gun deaths for each month of the year

month = [row[2] for row in data]
month_counts = {}

for row in month:
    if row not in month_counts:
        month_counts[row] = 1
    else:
        month_counts[row] += 1
print(month_counts)

{'06': 8677, '03': 8289, '04': 8455, '07': 8989, '01': 8273, '08': 8783, '11': 8243, '05': 8669, '12': 8413, '10': 8406, '09': 8508, '02': 7093}


In [30]:
# Exploring the number of homicides per month of the year

intents = [row[3] for row in data]
homicide_month_counts = {}
for i,mth in enumerate(month):
    if mth not in homicide_month_counts:
        homicide_month_counts[mth] = 0
    if intents[i] == "Homicide":
        homicide_month_counts[mth] += 1
print(homicide_month_counts)   

{'06': 3130, '03': 2780, '04': 2845, '07': 3269, '01': 2829, '08': 3125, '11': 2919, '05': 2976, '12': 3191, '10': 2968, '09': 2966, '02': 2178}


In [34]:
# Exploring the number of homicides according to the season of the year

summer = homicide_month_counts["06"] + homicide_month_counts["07"] + homicide_month_counts["08"]
print(summer)
spring = homicide_month_counts["03"] + homicide_month_counts["04"] + homicide_month_counts["05"]
print(spring)
fall = homicide_month_counts["09"] + homicide_month_counts["10"] + homicide_month_counts["11"]
print(fall)
winter = homicide_month_counts["12"] + homicide_month_counts["01"] + homicide_month_counts["02"]
print(winter)
summer_ratio = (summer-fall)/fall
print(summer_ratio)
summer_ratio2 = (summer-winter)/winter
print(summer_ratio2)

9524
8601
8853
8198
0.07579351632215069
0.16174676750426933


There are more homicides happening in the summer, between 7.5% and 16% more than other seasons.  There are less homicides in the winter months.